sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -566,6 +566,23 @@ package object dsl extends SQLConfHelper {
       }
 
       def deduplicate(colNames: Attribute*): LogicalPlan = Deduplicate(colNames, logicalPlan)
+
+      def watermark(expr: Expression, delayThreshold: CalendarInterval): LogicalPlan = {
+        val namedExpression = expr match {
+          case e: NamedExpression => e
+          case e: Expression => UnresolvedAlias(e)
```

Comment:
shall we fail here? It's a test and we should require the test to name the event time expr always